packages/data-context/test/unit/sources/GitDataSource.spec.ts

```diff
@@ -40,10 +63,18 @@ describe('GitDataSource', () => {
 
     process.chdir(projectPath)
 
+    const dfd = pDefer()
+
+    onGitInfoChange.onFirstCall().callsFake(dfd.resolve)
+
+    gitInfo.setSpecs([fooSpec, aRecordSpec, xhrSpec])
+
+    await dfd.promise
+
     const [created, unmodified, modified] = await Promise.all([
-      ctx.git.gitInfo(fooSpec),
-      ctx.git.gitInfo(aRecordSpec),
-      ctx.git.gitInfo(xhrSpec),
+      gitInfo.gitInfoFor(fooSpec),
+      gitInfo.gitInfoFor(aRecordSpec),
+      gitInfo.gitInfoFor(xhrSpec),
```

Comment:
I'm glad these tests were useful during this refactor - you hardly changed them which is a good sign, I think in general we are doing a good job of testing behaviours, not implementation details.